metrics.go

```diff
@@ -77,12 +77,24 @@ var (
 		Buckets: durationPerBlockHistogram,
 	})
 
+	fetchDurationCarSuccessMetric = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Name:    prometheus.BuildFQName("ipfs", "caboose", "fetch_duration_car_success"),
```

Comment:
We need `fetch_duration_car_peer_success` version as well (to match what we already have for blocks).
Even if we have no retries today, we should add these metrics from day one, so we can add them to Grafana.